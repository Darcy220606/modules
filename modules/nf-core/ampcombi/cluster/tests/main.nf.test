nextflow_process {

    name "Test Process AMPCOMBI_CLUSTER"
    script "../main.nf"
    process "AMPCOMBI_CLUSTER"

    tag "modules"
    tag "modules_nfcore"
    tag "ampcombi"
    tag "ampcombi/cluster"
    tag "ampcombi/complete"
    tag "ampcombi/parsetables"

    config "./nextflow.config"
    setup {
        run("AMPCOMBI_PARSETABLES", alias:"AMPCOMBI_PARSETABLES_1") {
            script "../../../ampcombi/parsetables/main.nf"
            process {
                """
                amp_input = [[id:'sample_1'],
                             [file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files_0.2/ampir/sample_1/sample_1.ampir.tsv', checkIfExists: true),
                              file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files_0.2/amplify/sample_1/sample_1.amplify.tsv', checkIfExists: true)]
                              ]
                faa_input = file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_faa_0.2/sample_1.faa', checkIfExists: true)
                gbk_input = file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_gbk_0.2/sample_1.gbff', checkIfExists: true)

                input[0] = amp_input
                input[1] = faa_input
                input[2] = gbk_input
                input[3] = []
                """
            }
        }
        run("AMPCOMBI_PARSETABLES", alias:"AMPCOMBI_PARSETABLES_2") {
            script "../../../ampcombi/parsetables/main.nf"
            process {
                """
                amp_input = [[id:'sample_2'],
                             [file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files_0.2/ampir/sample_2/sample_2.ampir.tsv', checkIfExists: true),
                              file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_files_0.2/amplify/sample_2/sample_2.amplify.tsv', checkIfExists: true)]
                              ]
                faa_input = file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_faa_0.2/sample_2.faa', checkIfExists: true)
                gbk_input = file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/ampcombi/test_gbk_0.2/sample_2.gbff', checkIfExists: true)

                input[0] = amp_input
                input[1] = faa_input
                input[2] = gbk_input
                input[3] = []
                """
            }
        }
        run("AMPCOMBI_COMPLETE") {
            script "../../../ampcombi/complete/main.nf"
            process {
                """
                ch_one = AMPCOMBI_PARSETABLES_1.out.tsv.map{ it[1] }
                ch_two = AMPCOMBI_PARSETABLES_2.out.tsv.map{ it[1] }

                ch_one
                    .mix( ch_two )
                    .collect()
                    .set { ch_input_for_ampcombi_complete }

                input[0] = ch_input_for_ampcombi_complete
                """
            }
        }

    }

    test("ampcombi_cluster - metagenome") {
        when {
            process {
                """
                input[0] = AMPCOMBI_COMPLETE.out.tsv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.cluster_tsv[0]).name,
                    file(process.out.rep_cluster_tsv[0]).name,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("ampcombi_cluster - metagenome - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = AMPCOMBI_COMPLETE.out.tsv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
