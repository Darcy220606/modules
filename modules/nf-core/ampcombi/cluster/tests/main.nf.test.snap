{
    "ampcombi_cluster - metagenome - stub": {
        "content": [
            {
                "0": [
                    "Ampcombi_summary_cluster.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "Ampcombi_summary_cluster_representative_seq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "Ampcombi_cluster.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "3": [
                    "versions.yml:md5,9e47adfcc53153d0eda32394f4198bb9"
                ],
                "cluster_tsv": [
                    "Ampcombi_summary_cluster.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "log": [
                    "Ampcombi_cluster.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "rep_cluster_tsv": [
                    "Ampcombi_summary_cluster_representative_seq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,9e47adfcc53153d0eda32394f4198bb9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-28T16:17:48.412966801"
    },
    "ampcombi_cluster - metagenome": {
        "content": [
            "Ampcombi_summary_cluster.tsv",
            "Ampcombi_summary_cluster_representative_seq.tsv",
            [
                "versions.yml:md5,9e47adfcc53153d0eda32394f4198bb9"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-28T16:17:32.546491818"
    }
}